packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/views/explorer/services/DefaultObjectDuplicator.java

```diff
@@ -0,0 +1,161 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.application.views.explorer.services;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.eclipse.emf.common.command.Command;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.EReference;
+import org.eclipse.emf.ecore.util.ECrossReferenceAdapter;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.emf.edit.command.AddCommand;
+import org.eclipse.emf.edit.command.SetCommand;
+import org.eclipse.emf.edit.domain.EditingDomain;
+import org.eclipse.sirius.components.collaborative.api.ChangeKind;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.emf.services.api.IEMFEditingContext;
+import org.eclipse.sirius.components.emf.utils.SiriusEMFCopier;
+import org.eclipse.sirius.components.representations.Failure;
+import org.eclipse.sirius.components.representations.IStatus;
+import org.eclipse.sirius.components.representations.Success;
+import org.eclipse.sirius.web.application.views.explorer.services.api.DuplicationSettings;
+import org.eclipse.sirius.web.application.views.explorer.services.api.IObjectDuplicator;
+import org.eclipse.sirius.web.domain.services.api.IMessageService;
+
+/**
+ * The default implementation if IObectDuplicator, used only if no more specific implementation applies.
+ *
+ * @author pcdavid
+ */
+public class DefaultObjectDuplicator implements IObjectDuplicator {
+    public static final String NEW_OBJECT = "newObject";
```

Comment:
Shouldn't the one in `DuplicateObjectEventHandler` be used? Or at least a single one maybe on `IObjectDuplicator`.